zipline/assets/assets.py

```diff
@@ -338,6 +446,8 @@ def spawn_asset(self, identifier, **kwargs):
         self.identifier_cache[identifier] = asset
         if asset.symbol is not '':
             self.sym_cache.setdefault(asset.symbol, []).append(asset)
+        if isinstance(asset, Future) and asset.root_symbol is not '':
+            self.future_cache.setdefault(asset.root_symbol, []).append(asset)
```

Comment:
To avoid an instance check, we could bump this up to be within the 'elif asset_type.lower() == 'future':' above.